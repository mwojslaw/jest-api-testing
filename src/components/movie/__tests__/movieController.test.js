const request = require("supertest");
const createServer = require("./../../../../server");
const { openMockConnection, getMockDB } = require("./../../../utils/mock");
const seed = require("./../../../utils/seed");

let connection;
let db;

beforeAll(async () => {
  connection = await openMockConnection();
  db = await getMockDB(connection);
  await seed(db);
});

afterAll(async () => {
  await connection.close();
});

it("/movies", async () => {
  const response = await request(createServer(() => db)).get("/movies");
  expect(response.status).toBe(200);
});
