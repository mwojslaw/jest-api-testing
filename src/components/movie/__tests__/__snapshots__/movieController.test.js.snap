// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/movies 1`] = `"{\\"movies\\":[{\\"_id\\":\\"5b6ed4db56471fe13b20ad72\\",\\"title\\":\\"Harry Potter and the philosopher's stone\\"},{\\"_id\\":\\"5b6ed4db56471fe13b20ad73\\",\\"title\\":\\"Harry Potter and the chamber of secrets\\"}]}"`;
